core/services/gateway/handlers/functions/allowlist.go

```diff
@@ -172,17 +196,50 @@ func (a *onchainAllowlist) updateFromContractV1(ctx context.Context, blockNum *b
 	if err != nil {
 		return errors.Wrap(err, "unexpected error during functions_allow_list.NewTermsOfServiceAllowList")
 	}
-	addrList, err := tosContract.GetAllAllowedSenders(&bind.CallOpts{
+
+	count, err := tosContract.GetAllowedSendersCount(&bind.CallOpts{
 		Pending:     false,
 		BlockNumber: blockNum,
 		Context:     ctx,
 	})
 	if err != nil {
-		return errors.Wrap(err, "error calling GetAllAllowedSenders")
+		return errors.Wrap(err, "unexpected error during functions_allow_list.GetAllowedSendersCount")
 	}
-	a.update(addrList)
+
+	allowedSenderList := make([]common.Address, 0)
+	var storageWg sync.WaitGroup
+	for idxStart := uint64(0); idxStart < count; idxStart += uint64(a.config.OnchainAllowlistBatchSize) {
+		idxEnd := idxStart + uint64(a.config.OnchainAllowlistBatchSize)
+		if idxEnd >= count {
+			idxEnd = count - 1
+		}
+
+		allowedSendersBatch, err := tosContract.GetAllowedSendersInRange(&bind.CallOpts{
```

Comment:
We might hit RPC client rate limiting if we just send a bunch of these range queries in a row. Consider spacing them in time, just like queryLoop() in subscriptions.go does.

Reply:
good catch, added a configurable throttleTicker that will prevent this.